reactive/openstack.py

```diff
@@ -1,18 +1,56 @@
+import collections
 from distutils.util import strtobool
 from charmhelpers.core import hookenv
+from charmhelpers.contrib.charmsupport import nrpe
 from charms.reactive import (
     hook,
+    when,
     when_all,
     when_any,
     when_not,
     is_flag_set,
     toggle_flag,
     clear_flag,
+    set_flag,
 )
+
 from charms.reactive.relations import endpoint_from_name
 
 from charms import layer
 
+# NOTE (rgildein): This namedtuple is used to define NRPE check. Composed of:
+#                  name - NRPE check shortname
+#                  interface - OpenStack interface to be check
+#                  config - option name defined in config.yaml
+#                  config_skip - option name define in config.yaml
+#                  all - if the option in config.yaml can be `all`
+NrpeCheck = collections.namedtuple(
+    "NRPE_CHECK", "name interface config config_skip all",
+    defaults=[None, False])
+
+NRPE_CHECKS = [
+    # check , OpenStack interface, check_all allowed, skip ids
+    NrpeCheck("kubernetes_subnet", "subnet", "subnet-id"),
+    NrpeCheck("kubernetes_network", "network", "floating-network-id"),
+    NrpeCheck("openstack_networks", "network", "nrpe-network-ids",
+              "nrpe-network-skip-ids", True),
+    NrpeCheck("openstack_subnets", "subnet", "nrpe-subnet-ids"),
+    NrpeCheck("openstack_ports", "port", "nrpe-port-ids", "nrpe-port-skip-ids",
+              True),
+    NrpeCheck("openstack_floating_ips", "floating-ip", "nrpe-floating-ip-ids",
+              "nrpe-floating-ip-skip-ids", True),
+    NrpeCheck("openstack_servers", "server", "nrpe-server-ids",
+              "nrpe-server-skip-ids", True),
+    NrpeCheck("openstack_security_groups", "security-group",
+              "nrpe-security-group-skip-ids"),
+]
+NRPE_OPENSTACK_INTERFACE = "check_openstack_interface.py"
+NRPE_CONFIG_FLAGS_CHANGED = [
+    *["config.changed.{}".format(c.config) for c in NRPE_CHECKS],
+    *["config.changed.{}".format(c.config_skip) for c in NRPE_CHECKS
+      if c.config_skip]
+]
+
```

Comment:
What would you think about moving all these constants to the "lib" folder? Something like `lib/nrpe_helpers.py` ?